episodes/chapter5.md

```diff
@@ -0,0 +1,302 @@
+---
+title: "Running a PET analysis using the python client"
+teaching: 
+exercises: 
+---
+
+:::::::::::::::::::::::::::::::::::::: questions
+- How to check the status of a given collaboration within vantage6, programatically, using the Python client?
+- How to request a task based on a given algorithm, programatically, through vantage6's Python client?
+
+:::::::::::::::::::::::::::::::::::::::::::::::::
+
+::::::::::::::::::::::::::::::::::::: objectives
+
+- Understand the programmatic approach for performing a data analysis through a given scenario.
+
+::::::::::::::::::::::::::::::::::::::::::::::::
+
+# Vantage6 Python client - basic concepts
+
+The vantage6 Python client is a library designed to facilitate interaction with the vantage6 server, to perform various tasks such as creating computation tasks, managing organizations, collaborations, users, and collecting results. It communicates with the server through its REST API, handling encryption and decryption of data for secure operations. The client aims to comprehensively support all aspects of server communication, making it a useful tool for users looking to leverage the full capabilities of the vantage6 platform programmatically.
+
+Creating an instance of the vantage6 Python client is relatively straightforward, requiring only (after installing the corresponding libraries) to define the server connection details: server URL, login credentials, and the organization's private key (if encryption is being used). For simplicity, these settings can be defined on a separate Python module like the following:
+
+``` python
+# config.py
+
+server_url = "https://<vantage6-server-address>" 
+server_port = 443 
+server_api = "/api" 
+
+username = "MY USERNAME"
+password = "MY PASSWORD"
+
+organization_key = "" # Path to the encryption key, if encryption is enabled
+```
+
+Once you have created the Python module with the configuration settings, you can create the client instance as follows:
+
+``` python
+from vantage6.client import UserClient as Client
+
+# Note: we assume here the config.py you just created is in the current directory.
+# If it is not, then you need to make sure it can be found on your PYTHONPATH
+import config
+
+# Initialize the client object, and run the authentication
+client = Client(config.server_url, config.server_port, config.server_api,
+                log_level='debug')
+client.authenticate(config.username, config.password)
+
+# Optional: setup the encryption, if you have an organization_key
```

Comment:
Currently this step is not optional. Even when no encryption is used you should trigger this line. However there is an issue open to change this: https://github.com/vantage6/vantage6/issues/1302

So it might be optional after all. Maybe good to add this issue in the comment above? So we dont forget / the participant is aware of the change?

Reply:
Updated for consistency with how it currently works: using an empty string when there is no organization key. A note about the Change Request was added.